From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 1 Jun 2021 22:00:00 +0000
Subject: [PATCH] glib: Fix type in poppler_rectangle_free

e3fed321f230 changed the allocation to use PopplerRectangleExtended. We
need to change the free to match, otherwise the slice allocator can get
confused.

Fixes https://gitlab.freedesktop.org/poppler/poppler/-/issues/1087
---
 glib/poppler-page.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/glib/poppler-page.cc b/glib/poppler-page.cc
index 7536cfcb..e81c1e12 100644
--- a/glib/poppler-page.cc
+++ b/glib/poppler-page.cc
@@ -1663,7 +1663,8 @@ PopplerRectangle *poppler_rectangle_copy(PopplerRectangle *rectangle)
  */
 void poppler_rectangle_free(PopplerRectangle *rectangle)
 {
-    g_slice_free(PopplerRectangle, rectangle);
+    auto ext_rectangle = reinterpret_cast<PopplerRectangleExtended *>(rectangle);
+    g_slice_free(PopplerRectangleExtended, ext_rectangle);
 }
 
 /**
